### PR TITLE
Audits `XrpSignerListSet` Protobuf Wrapper

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

This PR adds additional logic validations and descriptive error throwing to the class `XrpSignerListSet`.

### Context of Change

Another component of our attempt to make protobuf wrapper classes as safe and informative as possible. An SDK safety net.

### Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)

## Before / After

`XrpSignerListSet` performs logic/error checking, based on the business logic in the docs.

## Test Plan

New tests added to confirm that errors are correctly thrown. Existing tests were edited to properly account for the new errors. CI passes.

## Future Tasks
This is part of a series, with all subtasks listed here: https://app.asana.com/0/1188384817180008/1193021408531788/f

### CODE REVIEW

File: src/XRP/model/xrp-signer-list-set.ts [hunk 1]
```diff
@@ -20,14 +21,42 @@ export default class XrpSignerListSet {
   public static from(
     signerListSet: SignerListSet,
     xrplNetwork: XrplNetwork,
-  ): XrpSignerListSet | undefined {
+  ): XrpSignerListSet {
     const signerQuorum = signerListSet.getSignerQuorum()?.getValue()
-    if (!signerQuorum) {
-      return undefined
+    if (signerQuorum == undefined) {
```
Comment (tedkalaw) [inline src/XRP/model/xrp-signer-list-set.ts:26]: `===`

File: src/XRP/model/xrp-signer-list-set.ts [hunk 2]
```diff
@@ -20,14 +21,42 @@ export default class XrpSignerListSet {
...
       .getSignerEntriesList()
       .map((signerEntry) => XrpSignerEntry.from(signerEntry, xrplNetwork))
+    if (signerQuorum != 0) {
+      if (signerEntries.length == 0) {
+        throw new XrpError(
+          XrpErrorType.MalformedProtobuf,
+          'SignerListSet protobuf does not contain `SignerEntries` field with nonzero `SignerQuorum` field.',
+        )
+      }
+      if (signerEntries.length > 8) {
```
Comment (tedkalaw) [inline src/XRP/model/xrp-signer-list-set.ts:42]: const for this value? could be a file-scoped const

File: src/XRP/model/xrp-signer-list-set.ts [hunk 3]
```diff
@@ -20,14 +21,42 @@ export default class XrpSignerListSet {
...
+        throw new XrpError(
+          XrpErrorType.MalformedProtobuf,
+          'SignerListSet protobuf has greater than 8 members in the `SignerEntries` field.',
+        )
+      }
+      const accounts: string[] = []
+      signerEntries.forEach((signerEntry) => {
+        const signerAccount = signerEntry.accountXAddress
```
Comment (tedkalaw) [inline src/XRP/model/xrp-signer-list-set.ts:50]: extreme naming nit - `signerAccountXAddress`?

File: src/XRP/model/xrp-signer-list-set.ts [hunk 4]
```diff
@@ -20,14 +21,42 @@ export default class XrpSignerListSet {
...
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'SignerListSet protobuf does not contain `SignerQuorum` field.',
+      )
     }
     const signerEntries = signerListSet
       .getSignerEntriesList()
       .map((signerEntry) => XrpSignerEntry.from(signerEntry, xrplNetwork))
+    if (signerQuorum != 0) {
+      if (signerEntries.length == 0) {
```
Comment (tedkalaw) [inline src/XRP/model/xrp-signer-list-set.ts:36]: `===`

File: src/XRP/model/xrp-signer-list-set.ts [hunk 5]
```diff
@@ -20,14 +21,42 @@ export default class XrpSignerListSet {
...
+          'SignerListSet protobuf does not contain `SignerEntries` field with nonzero `SignerQuorum` field.',
+        )
+      }
+      if (signerEntries.length > 8) {
+        throw new XrpError(
+          XrpErrorType.MalformedProtobuf,
+          'SignerListSet protobuf has greater than 8 members in the `SignerEntries` field.',
+        )
+      }
+      const accounts: string[] = []
```
Comment (tedkalaw) [inline src/XRP/model/xrp-signer-list-set.ts:48]: can we use a `Set` here instead?

**Review by tedkalaw — CHANGES_REQUESTED** [review]
looks p good, some thoughts inline

**Review by tedkalaw — APPROVED** [review]

File: src/XRP/model/xrp-signer-list-set.ts [hunk 6]
```diff
@@ -20,14 +21,43 @@ export default class XrpSignerListSet {
...
     xrplNetwork: XrplNetwork,
-  ): XrpSignerListSet | undefined {
+  ): XrpSignerListSet {
     const signerQuorum = signerListSet.getSignerQuorum()?.getValue()
-    if (!signerQuorum) {
-      return undefined
+    if (signerQuorum === undefined) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'SignerListSet protobuf does not contain `SignerQuorum` field.',
```
Comment (amiecorso) [inline src/XRP/model/xrp-signer-list-set.ts:29]: lowercase

**Review by amiecorso — APPROVED** [review]

**Pull request merged**